decoder/scanarea.go

```diff
@@ -0,0 +1,74 @@
+package decoder
+
+import (
+	"golbat/config"
+	"golbat/geo"
+	"strings"
+)
+
+type ScanParameters struct {
+	ProcessPokemon   bool
+	ProcessWild      bool
+	ProcessNearby    bool
+	ProcessWeather   bool
+	ProcessPokestops bool
+	ProcessGyms      bool
+	ProcessCells     bool
+}
+
+func FindScanConfiguration(scanContext string, lat, lon float64) ScanParameters {
+	var areas []geo.AreaName
+	areaLookedUp := false
+
+	for _, rule := range config.Config.ScanRules {
+		if len(rule.AreaNames) > 0 {
+			if !areaLookedUp {
+				areas = MatchStatsGeofence(lat, lon)
+				areaLookedUp = true
+			}
+			if !geo.AreaMatchWithWildcards(areas, rule.AreaNames) {
+				continue
+			}
+		}
+		if len(rule.ScanContext) > 0 {
+			found := false
+			for _, context := range rule.ScanContext {
+				if strings.EqualFold(context, scanContext) {
+					found = true
+					break
+				}
+			}
+			if !found {
+				continue
+			}
+		}
+
+		// We have a match
+
+		defaultTrue := func(value *bool) bool {
+			if value == nil {
+				return true
+			}
+			return *value
+		}
```

Comment:
Move it outside loop

Reply:
…because?

Reply:
Why would you want function declaration Inside loop, when you can initialize it once near variables?

Reply:
Because 1) it’s not in a loop that will continue. It’s literally followed by a return statement. 
2) compiler optimisation would mean it is not actually created each loop even if it was. 
3) it is clearer code.

Reply:
1-2. You are right.
3. Seriously it is?

Before
![before](https://user-images.githubusercontent.com/10072920/229349338-82d0296e-4d58-41f8-a267-a13290f105d4.png)

After
![after](https://user-images.githubusercontent.com/10072920/229349342-b140e560-bc3a-4869-b5b8-9c82ed36a32f.png)